akka-stream-tests/src/test/scala/akka/stream/impl/GraphStageLogicSpec.scala

```diff
@@ -155,7 +164,13 @@ class GraphStageLogicSpec extends StreamSpec with GraphInterpreterSpecKit {
 
       Source.empty.via(g).runWith(TestSink.probe)
         .request(9)
-        .expectNextN(1 to 8)
+        .expectNext(1)
+        //emitting with callback gives nondeterminism whether 2 or 3 will be pushed first
+        .expectNextUnordered(2, 3)
+        .expectNext(4)
+        .expectNext(5)
+        .expectNextUnordered(6, 7)
+        .expectNext(8)
```

Comment:
Would these tests cases fail non-deterministically before the changes introduced in `GraphStage.scala`?